flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -508,7 +508,11 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
 			throw new YarnDeploymentException("Could not retrieve information about free cluster resources.", e);
 		}
 
-		final int yarnMinAllocationMB = yarnConfiguration.getInt(YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB, 0);
+		final int yarnMinAllocationMB = yarnConfiguration.getInt(
+				YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB,
+				YarnConfiguration.DEFAULT_RM_SCHEDULER_MINIMUM_ALLOCATION_MB);
+		Preconditions.checkArgument(yarnMinAllocationMB > 0, "The minimum allocation memory "
+				+ "(" + yarnMinAllocationMB + " MB) configured via 'yarn.scheduler.minimum-allocation-mb' should be greater than 0.");
```

Comment:
The string literal can be replaced with `YarnConfiguration.RM_SCHEDULER_MINIMUM_ALLOCATION_MB`.